ios-target/View/Home/ModalViewController.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  TargetForm.swift
+//  ios-target
+//
+//  Created by sol manini on 6/17/19.
+//  Copyright © 2019 TopTier labs. All rights reserved.
+//
+
+import UIKit
+
+class ModalViewController: UIViewController {
+  
+  lazy var blackView: UIView = {
+    let view = UIView()
+    view.translatesAutoresizingMaskIntoConstraints = false
+    view.alpha = 0
+    return view
+  }()
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    view.insertSubview(blackView, at: 0)
+    
+    NSLayoutConstraint.activate([
+      view.leadingAnchor.constraint(equalTo: blackView.leadingAnchor),
+      view.trailingAnchor.constraint(equalTo: blackView.trailingAnchor),
+      view.widthAnchor.constraint(equalTo: blackView.widthAnchor),
+      view.heightAnchor.constraint(equalTo: blackView.heightAnchor)
+      ])
+    
+    blackView.backgroundColor = UIColor(white: 0, alpha: 0.5)
+    blackView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(handleDismiss)))
+  }
+  
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    dimBackground(isSet: false)
+  }
+  
+  func dimBackground(isSet: Bool = false, completion: ((Bool) -> Void)? = nil) {
+    UIView.animate(withDuration: isSet ? 0.2 : 0.3,
```

Comment:
German is teaching you the ancient art of avoiding weak self in animations? haha

Reply:
I guess you mean :flag-cu: I prefer to weakify to keep consitence 🙂

Reply:
In this case it is not needed. The animation block is not referenced by self so no retain cycle. self will be retained until the animation finishes though.

Reply:
what may happen is that the self instance maybe destroyed during the animation, since you're not retaining it, and since you're not weakyfying (so self is not optional) you may crash.

Reply:
The instance will not be destroyed. You have to trust ARC man 😄 
Since the block is referencing self, the references count will be at least 1 and the instance will not be dealloced.

Reply:
I remember dealing with some crashes related to this but I don't remember what it was very clearly 😖